packages/settings-library/src/settings_library/node_ports.py

```diff
@@ -0,0 +1,46 @@
+from typing import ClassVar, Final
+
+from models_library.basic_types import PortInt
+from pydantic import Field, NonNegativeInt, PositiveInt, SecretStr
+
+from ._constants import MINUTE
+from .base import BaseCustomSettings
+from .postgres import PostgresSettings
+from .storage import StorageSettings
+
+NODE_PORTS_400_REQUEST_TIMEOUT_ATTEMPTS_DEFAULT_VALUE: Final[NonNegativeInt] = 3
+
+
+class StorageAuthSettings(BaseCustomSettings):
+    NODE_PORTS_STORAGE_HOST: str
+    NODE_PORTS_STORAGE_PORT: PortInt
+
+    NODE_PORTS_STORAGE_LOGIN: str
+    NODE_PORTS_STORAGE_PASSWORD: SecretStr
+
+    @property
+    def base_url(self) -> str:
+        return StorageSettings(
+            STORAGE_HOST=self.NODE_PORTS_STORAGE_HOST,
+            STORAGE_PORT=self.NODE_PORTS_STORAGE_PORT,
+        ).base_url
+
+    class Config:
+        json_encoders: ClassVar = {
+            SecretStr: lambda v: v.get_secret_value() if v else None,
+        }
+
+
+class NodePortsSettings(BaseCustomSettings):
+    NODE_PORTS_STORAGE: StorageSettings = Field(auto_default_from_env=True)
```

Comment:
You do not need StorageAuthSettings here?

Reply:
it's on the line below. We keep both storage endpoints. If `NODE_PORTS_STORAGE_AUTH` is provided, it will be used in favour of `NODE_PORTS_STORAGE`

Reply:
Aha so you have both of them!  I did not notice . So, what happens when both are captured? 

Would it not make more sense to inherit `StorageAuthSettings(StorageSettings)` and make LOGIN (i.e. USERNAME) and PASSWORD optional?

Reply:
There is a similar scenario with `SMTPSettings`. It is far from ideal but i think it is a good example of the mindset i am proposing